### PR TITLE
Feature/handle properties of complex metadata types

### PR DESCRIPTION
**Pull request by froucher**
Todo List:
- [x] Compact Layout Fix
- [x] Add current changeset integration tests to functional tests
- [x] SearchLayout test
- [x] Add more tests
- [x] Move to general package.json
- [x] Optionally functional tests without deploy
- [x] README.md explaining how to work with functional tests
- [x] Create a test script for travis
- [x] ActionOverride test
- [x] More tests adding, updating, removing multiple metadata
- [x] Internal Tests
- [x] Remove some now redundant tests from `test-integration`
- [x] Run functional tests in CI

+@amtrack, thank you for your support!

### CODE REVIEW

File: lib/describe-metadata-service.js [hunk 1]
```diff
@@ -41,7 +41,8 @@ var childTypes = [{
 	xmlName: 'CompactLayout',
 	tagName: 'compactLayouts',
 	parent: 'CustomObject',
-	key: 'fullName'
+	key: 'fullName',
+	isNamed: true
```
Comment (froucher) [inline lib/describe-metadata-service.js:45]: This is the compact layout fix

Comment (amtrack) [reply lib/describe-metadata-service.js:45]: @froucher Could you please create a separate PR for the CompactLayout fix? Then we can merge it right away.

Comment (froucher) [reply lib/describe-metadata-service.js:45]: This is the new PR #207 

Thanks!

Comment (froucher) [reply lib/describe-metadata-service.js:46]: Added SharingReason and BusinessProcess similar fix to the PR #232

**Comment by amtrack** [comment]
@froucher This looks really really good! 😮 ❤️ 

I just did a first round of review and will continue soon.
You can cherry-pick from this branch https://github.com/amtrack/force-dev-tool/commits/feature/handle-properties-of-complex-metadata-types if you want.

**Comment by amtrack** [comment]
@froucher I've continued my review and pushed some more commits to my [feature/handle-properties-of-complex-metadata-types](https://github.com/amtrack/force-dev-tool/commits/feature/handle-properties-of-complex-metadata-types) branch, please cherry-pick or merge.

Besides those commits, I think this PR can be merged.

For debugging, running the test suite with `npm run test:functional:doing` and changing into the temp directory which is git controlled is a game changer!!! 🎆 

Well done! 👏 🙏

**Comment by froucher** [comment]
Good job @amtrack! 👏 👍 

Commits has been cherry-picked and travis continues green!

Thank you.

**Comment by amtrack** [comment]
@froucher ¡Muchas gracias!
This is really great work!

**Pull request merged**